packages/twenty-front/src/modules/object-record/record-field/types/FieldMetadata.ts

```diff
@@ -288,14 +287,17 @@ type FieldActorSource =
   | 'SYSTEM'
   | 'WORKFLOW';
 
-export type FieldActorValue = {
-  source: FieldActorSource;
-  workspaceMemberId: string | null;
-  name: string;
-  context: {
-    provider?: ConnectedAccountProvider;
-  } | null;
-};
+export const actorSchema = z.object({
+  source: z.string(),
+  workspaceMemberId: z.string().nullable(),
+  name: z.string(),
+  context: z
+    .object({
+      provider: z.nativeEnum(ConnectedAccountProvider).optional(),
+    })
+    .nullable(),
+});
+export type FieldActorValue = z.infer<typeof actorSchema>;
```

Comment:
@lucasbordeau what do you think of such a gymnastic ? 
I feel like we have several duplicated type declaration between `type` and zod `schemas`, we could factorize them in order to avoid desyncrhonization